package.json

```
@@ -18,7 +18,10 @@
   },
   "scripts": {
     "test": "mocha test && standard",
-    "preinstall": "node build.js"
+    "build": "node build.js",
+    "submodule": "git submodule update --recursive --init",
+    "prepublishOnly": "npm run submodule && npm run build",
+    "postinstall": "npm run build"
```

I'm still not convinced this is correct, `preinstall` scripts run when the package is a dependency. I checked this with the following steps:

```
$ mkdir test && cd test
$ cat > package.json
{
  "dependencies": {
    "minidump": "*"
  }
}
^D
$ npm install
```

The `preinstall` script ran and the module was compiled.

```suggestion
    "preinstall": "node build.js"
```

---

I'll commit the dump_syms on macOS code for now and we can come back to the submodule/install stuff later if we need to.

---

- At the minimum, you should not remove prepublishOnly! That way, you or someone else will forgot to instantiate submodules, and the package will not have the required files

- preinstall works but it does not make sense here. It works because minidump does not use any dependency in build.js. postinstall is more general. It allows bootstrapping the dependencies, and then run the build.